File: app/src/main/java/com/smedialink/abakarmagomedov/githubapplication/BaseActivity.java
package com.smedialink.abakarmagomedov.githubapplication;

import android.app.ProgressDialog;
import android.content.Intent;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;

/**
 * Created by abakarmagomedov on 15/05/17.
 */

public class BaseActivity extends AppCompatActivity {

    protected @NonNull ProgressDialog mProgressDialog;

    protected void start(Class<?> cls, String login) {
        Intent intent = new Intent(this, cls);
        intent.putExtra(Constants.LOGIN, login);
        startActivity(intent);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mProgressDialog = new ProgressDialog(this, R.style.ProgressBarStyle);
        mProgressDialog.setCancelable(false);
    }

    public void showProgress() {
        if (!mProgressDialog.isShowing()) {
            mProgressDialog.show();
        }

    }

    public void hideProgress() {
        if (mProgressDialog.isShowing()) {
            mProgressDialog.dismiss();
        }
    }


    protected AlertDialog getErrorDialog() {
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        builder.setPositiveButton(R.string.Ok, (dialog, which) -> dialog.dismiss());
        builder.setMessage(R.string.no_internet);
        return builder.create();
    }

}

File: app/src/main/java/com/smedialink/abakarmagomedov/githubapplication/BaseActivity.kt
package com.smedialink.abakarmagomedov.githubapplication

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.ProgressBar

/**
 * Created by abakarmagomedov on 15/05/17.
 */
abstract class BaseActivity : AppCompatActivity() {
    abstract var mProgressBar: ProgressBar?

    protected fun start(cls: Class<*>?, login: String?) {
        val intent = Intent(this, cls)
        intent.putExtra(Constants.LOGIN, login)
        startActivity(intent)
    }

    fun showProgress() {
        if (mProgressBar?.visibility != View.VISIBLE) {
            mProgressBar?.setVisibility(View.VISIBLE)
        }
    }

    fun hideProgress() {
        if (mProgressBar?.visibility == View.VISIBLE) {
            mProgressBar?.setVisibility(View.GONE)
        }
    }

    protected val errorDialog: AlertDialog
        protected get() {
            val builder = AlertDialog.Builder(this)
            builder.setPositiveButton(R.string.Ok) { dialog, which -> dialog.dismiss() }
            builder.setMessage(R.string.no_internet)
            return builder.create()
        }
}
